integration-tests/smoke/ccip/ccip_fees_test.go

```diff
@@ -101,11 +101,11 @@ func setupTokens(
 
 func Test_CCIPFees(t *testing.T) {
 	t.Parallel()
-	tenv := changeset.NewMemoryEnvironmentWithJobsAndContracts(t, logger.TestLogger(t), memory.MemoryEnvironmentConfig{
-		Chains:     2,
-		Nodes:      4,
-		Bootstraps: 1,
-	}, nil)
+	tenv, _ := testsetups.NewIntegrationEnvironment(
+		t,
+		changeset.WithMultiCall3(),
+		changeset.WithEnvironmentType(changeset.Memory),
```

Comment:
I wonder whether that should be driven by the code or some sort of env variable. I think all of the tests using `testsetups.NewIntegrationEnvironment` should be executable with both `inmem` and `docker` env. That way, you don't need to recompile everything to switch the env. I can imagine sth like this (don't focus on the names or the impl details, it's to show the high level ideas)


```go
func Test_Sth(t *testing.T) {
	tenv, _ := testsetups.NewIntegrationEnvironment(
		t,
		changeset.WithMultiCall3(),
		changeset.WithEnvironmentType(changeset.EnvDriven) // probably needs a better name
}


func NewIntegrationEnvironment(t *testing.T, opts ...changeset.TestOps) {
       // ...
       if testCfg.Type == changeset.EnvDriven {
             envType := os.Getenv("CCIP_INT_TESTS_ENV")
             if envType == "docker"{
                  return docker
             } else { // default to inmem
                  return inmem 
             }
      }
```

Reply:
That enables a couple of nice things:
* fast tests on the CI checks, but we can run everything in docker upon merge to main (it requires only a different ENV variable in CI configuration)
* doesn't require re-compilation if you want to debug locally using inmem instead of the docker hardcoded

Reply:
What's stops us from doing something like this in the tests:

```golang
func Test_Something(t *testing.T) {
  tenv, _ := testsetups.NewIntegrationEnvironment(
		t,
		changeset.WithMultiCall3(),
                 changeset.WithEnvironmentType(testhelpers.MustGetEnvTypeOrDefault(t)),
  )
  // .. rest of test code
}

// in package testhelpers
func MustGetEnvTypeOrDefault(t *testing.T) EnvType {
  envType := os.Getenv("CCIP_V16_TEST_ENV")
  if envType == "" || envType == "in-memory" {
    return EnvTypeMemory
  } else if envType == "docker" {
    return EnvTypeDocker
  }
  
  t.Fatalf("env var CCIP_V16_TEST_ENV must be either 'in-memory' or 'docker', defaults to 'in-memory' if unset, got: %s", envType)
}

```